spring-ws-core/src/main/java/org/springframework/ws/soap/addressing/version/Addressing10.java

```diff
@@ -48,7 +48,8 @@ public boolean hasRequiredProperties(MessageAddressingProperties map) {
 		if (map.getAction() == null) {
 			return false;
 		}
-		if (map.getReplyTo() != null || map.getFaultTo() != null) {
+		if ((map.getReplyTo() != null && !hasAnonymousAddress(map.getReplyTo()) && !hasNoneAddress(map.getReplyTo()))  
+				|| (map.getFaultTo() != null && !hasAnonymousAddress(map.getFaultTo()) && !hasNoneAddress(map.getFaultTo()))){  
 			return map.getMessageId() != null;
```

Comment:
Reading six boolean expressions chained together in collection of "ands", "ors" and "nots" takes me back to logical gate construction and makes me need a Venn diagram to grok. Could you extract some functions that embody the intent so its more readable?